storage/rocksdb/ha_rocksdb.cc

```diff
@@ -3656,8 +3668,17 @@ static int rocksdb_init_func(void *const p) {
       (rocksdb::BlockBasedTableOptions::IndexType)rocksdb_index_type;
 
   if (!rocksdb_tbl_options->no_block_cache) {
-    rocksdb_tbl_options->block_cache =
+    std::shared_ptr<rocksdb::Cache> block_cache =
         rocksdb::NewLRUCache(rocksdb_block_cache_size);
+    if (rocksdb_sim_cache_size > 0) {
```

Comment:
Rather than a new size for simulated block cache, maybe reuse block_cache_size and add an enum that controls the type of block_cache being created? 0 for LRU, 1 for simulated_cache, 2+ are reserved for new block_cache types? Then you don't end up creating a block_cache and throwing it away.